.github/workflows/push-to-docker-hub.yml

```diff
@@ -0,0 +1,41 @@
+name: Push to Docker Hub
+
+on:
+  push:
+    branches: [master]
+
+env:
+  IMAGE_FILE: image.tar
+
+jobs:
+  Push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Retrieve cached image
+        uses: actions/cache@v2
+        env:
+          cache-name: docker-images
+        with:
+          path: ${{ env.IMAGE_FILE }}
+          key: inspirehep-hep-${{ hashFiles('backend/poetry.lock') }}-${{ github.sha }}
```

Comment:
are you sure this works? The sha might change between the PR and the commit on master.

Reply:
Currently we rebase only. Does it change on rebase too?

Reply:
yes, in general it does